src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/RawAtlasGenerator.java

```diff
@@ -166,8 +165,14 @@ public RawAtlasGenerator withMetaData(final AtlasMetaData metaData)
     private Atlas buildRawAtlas()
     {
         final Time parseTime = Time.now();
-        connectOsmPbfToPbfConsumer(this.pbfReader);
-        this.reader.run();
+        try (CloseableOsmosisReader reader = connectOsmPbfToPbfConsumer(this.pbfReader))
```

Comment:
Very good!

Reply:
Kudos to @AlexHsieh22 for pointing this out in #83!